src/GraphQL.Tests/Types/Collections/SchemaTypesTests.cs

```diff
@@ -13,12 +13,20 @@ public class SchemaTypesTests
         public void does_not_request_instance_more_than_once()
         {
             // configure DI provider
-            var services = new SimpleContainer();
-            services.Singleton<StarWarsData>();
+            var services = new ServiceCollection();
+            services.AddSingleton<StarWarsData>();
+            services.AddSingleton<StarWarsQuery>();
+            services.AddSingleton<StarWarsMutation>();
+            services.AddSingleton<HumanType>();
+            services.AddSingleton<HumanInputType>();
+            services.AddSingleton<DroidType>();
+            services.AddSingleton<CharacterInterface>();
+            services.AddSingleton<EpisodeEnum>();
+            using var provider = services.BuildServiceProvider();
 
             // mock it so we can verify behavior
             var mock = new Mock<IServiceProvider>(MockBehavior.Loose);
-            mock.Setup(x => x.GetService(It.IsAny<Type>())).Returns<Type>(type => services.Get(type));
+            mock.Setup(x => x.GetService(It.IsAny<Type>())).Returns<Type>(type => provider.GetService(type));
 
             // run test
             var schema = new StarWarsSchema(mock.Object);
```

Comment:
I would guess that we can change the code to just return `provider` now where it was using `mock.Object`.

Reply:
`
  Moq.MockException : 
    Expected invocation on the mock once, but was 0 times: x => x.GetService(GraphQL.StarWars.StarWarsQuery)
    
    Performed invocations:
    
       Mock<IServiceProvider:1> (x):
       No invocations performed.
    
`

Reply:
without looking at the code i'm not sure how it's used, so i assume you've got it figured out.

Reply:
Mock verifies calls made through it so mock should be used, not real provider.